src/Controller/EnMarche/CitizenActionManagerController.php

```diff
@@ -82,4 +105,151 @@ public function editAction(Request $request, CitizenProject $project, CitizenAct
+                'slug' => $citizenAction->getSlug(),
+            ]);
+        }
+
+        $participants = $citizenActionManager->populateRegistrationWithAdherentsInformations($registrations);
+        $exported = $this->get(CitizenActionParticipantsExporter::class)->export($participants);
+
+        return new Response($exported, 200, [
+            'Content-Type' => 'text/csv',
+            'Content-Disposition' => 'attachment; filename="inscrits-a-l-action-citoyenne.csv"',
+        ]);
+    }
+
+    /**
+     * @Route("/{slug}/participants/contacter", name="app_citizen_action_contact_participants")
+     * @Method("POST")
+     * @Security("is_granted('EDIT_CITIZEN_ACTION', project)")
+     */
+    public function contactParticipantsAction(Request $request, CitizenAction $citizenAction, CitizenProject $project, CitizenActionManager $citizenActionManager): Response
+    {
+        $registrations = $this->getRegistrations($request, $citizenAction, self::ACTION_CONTACT);
+
+        if (0 == $registrations->count()) {
+            return $this->redirectToRoute('app_citizen_action_list_participants', [
+                'slug' => $citizenAction->getSlug(),
+            ]);
+        }
+
+        $command = new CitizenActionContactParticipantsCommand($registrations->toArray(), $this->getUser());
+
+        $form = $this->createForm(ContactMembersType::class, $command, ['csrf_token_id' => 'citizen_action.contact_participants'])
+            ->add('submit', SubmitType::class)
+            ->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->get(CitizenActionContactParticipantsCommandHandler::class)->handle($command);
+            $this->addFlash('info', $this->get('translator')->trans('citizen_action.contact.success'));
+
+            return $this->redirectToRoute('app_citizen_action_list_participants', [
+                'slug' => $citizenAction->getSlug(),
+            ]);
+        }
+
+        $uuids = array_map(function (EventRegistration $registration) {
+            return $registration->getUuid()->toString();
+        }, $registrations->toArray());
+
+        return $this->render('citizen_action_manager/contact_participants.html.twig', [
+            'citizen_action' => $citizenAction,
+            'citizen_project' => $project,
+            'contacts' => $uuids,
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/{slug}/participants/imprimer", name="app_citizen_action_print_participants")
+     * @Method("POST")
+     * @Security("is_granted('EDIT_CITIZEN_ACTION', project)")
+     */
+    public function printParticipantsAction(Request $request, CitizenAction $citizenAction, CitizenProject $project, CitizenActionManager $citizenActionManager): Response
+    {
+        $registrations = $this->getRegistrations($request, $citizenAction, self::ACTION_PRINT);
+
+        if (0 == $registrations->count()) {
```

Comment:
This code is really redundant. Perhaps it can be easily isolated in a private method of the controller ? 

Reply:
It's not useful, if we create a new method it will need a new `if` to do or not the redirection.